### PR TITLE
Add bandit B110 test and fix

### PR DESCRIPTION
**Pull request by edgarriba**
### Description

Fixes https://github.com/kornia/kornia/pull/1228#issuecomment-907122033

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

**Comment by edgarriba** [comment]
@cclauss

File: kornia/feature/hardnet.py [hunk 1]
```diff
@@ -148,10 +148,8 @@ def __init__(self, pretrained: bool = False):
     def weights_init(m):
         if isinstance(m, nn.Conv2d):
             nn.init.orthogonal_(m.weight.data, gain=0.6)
-            try:
+            if m.bias is not None:
```
Comment (edgarriba) [inline kornia/feature/hardnet.py:151]: @ducha-aiki hope this is not an issue

File: .github/workflows/tests_cuda.yml [hunk 2]
```diff
@@ -21,6 +21,7 @@ jobs:
     - name: Install dependencies
       shell: bash -l {0}
       run: |
+        apt-get install git
```
Comment (cclauss) [inline .github/workflows/tests_cuda.yml:24]: I have not seen the need to do this before but perhaps is caused by line 22.

Comment (edgarriba) [reply .github/workflows/tests_cuda.yml]: we need git in case someone is installing the package from source. See: https://github.com/kornia/kornia/blob/master/setup.py#L67

In the cpu workflow it's not needed because we create the environment installing the dependencies by hand - but in the case of cuda we reuse pytorch docker image which apparently doesn't include git.


Not sure if that's what you meant :)

Comment (cclauss) [reply .github/workflows/tests_cuda.yml]: I meant that `git` was always present in GitHub Actions.

Comment (edgarriba) [reply .github/workflows/tests_cuda.yml]: ah! mmm, then something is wrong from our side

Comment (edgarriba) [reply .github/workflows/tests_cuda.yml]: @cclauss I checked in local as well and the pytorch image we use doesn't include git. Was reading the dockerfile but quite not sure were and why they remove it: https://github.com/pytorch/pytorch/blob/master/Dockerfile
